File: gitea_repos_master/whatsapp/whatsapp-baileys-api/src/logic/mongo/fields.ts
export type PeriodFormats = 'seconds' | 'minutes' | 'hours';
export type Period = Partial<Record<PeriodFormats, number>>;

export class MongoFields {
  private _nextUse: string;
  private _lastUse: string;
  private _active: string;
  private _session: string;
  private _blockTime: Period;
  private _lockTime: Period;
  private _defaultFilter: Partial<Record<string, any>> | null;
  private _projection: Partial<Record<string, any>> | null;
  private _nextUseDelay: number;
  public defaultSort: any;

  constructor() {
    this._nextUse = 'next_use';
    this._lastUse = 'last_use';
    this._active = 'active';
    this._session = 'session';

    this.blockTime = { hours: 8 };
    this.lock_time = { minutes: 10 };

    this.nextUseDelay = 1;

    this.defaultFilter = { [this._active]: true };
    this.defaultSort = [{ [this.lastUse]: 1 }];

    this._projection = { [this._session]: 1 };
  }

  public get nextUse(): string {
    return this._nextUse;
  }

  public set nextUse(value: string) {
    this._nextUse = value;
  }

  public get active(): string {
    return this._active;
  }

  public set active(value: string) {
    this._active = value;
  }

  public get session(): string {
    return this._session;
  }

  public set session(value: string) {
    this._session = value;
  }

  public get lastUse(): string {
    return this._lastUse;
  }

  public set lastUse(value: string) {
    this._lastUse = value;
  }

  public get blockTime(): Period {
    return this._blockTime;
  }

  public set blockTime(value: Period) {
    this._blockTime = value;
  }

  public get lock_time(): Period {
    return this._lockTime;
  }

  public set lock_time(value: Period) {
    this._lockTime = value;
  }

  public get defaultFilter(): Partial<Record<string, any>> | null {
    return this._defaultFilter;
  }

  public set defaultFilter(
    filter_options: Partial<Record<string, any>> | null
  ) {
    this._defaultFilter = filter_options;
  }

  public get projection(): Partial<Record<string, any>> | null {
    return this._projection;
  }

  public set projection(options: Partial<Record<string, any>> | null) {
    this._projection = options;
  }

  public get nextUseDelay(): number {
    return this._nextUseDelay;
  }

  public set nextUseDelay(value: number) {
    this._nextUseDelay = value >= 0 ? value : 1;
  }
}

File: gitea_repos_master/isphere-classes/mongo-client/mongo_client/fields.py
from datetime import timedelta
from typing import Dict, Literal, Optional, Union

PeriodFormats = Literal[
    "days", "seconds", "microseconds", "milliseconds", "minutes", "hours", "weeks"
]
Period = Dict[Union[PeriodFormats, str], Union[int, float]]


class MongoFields:
    """Вспомогательный класс для хранения значений по умолчанию класса клиента Mongo"""

    def __init__(self):
        self.next_use = "next_use"
        self.last_use = "last_use"
        self.active = "active"
        self.session = "session"

        self.block_time: Period = {"hours": 8}
        self.lock_time: Period = {"minutes": 10}

        self.next_use_delay = 1

        self.default_filter = {self.active: True}
        self.default_sort = [(self.last_use, 1)]
        self.projection = {self.session: 1}

    @property
    def next_use(self) -> str:
        """Геттер на получение имени поля, отвечающего за timestamp блокировки сессии.

        :return: имя поля
        """
        return self._next_use

    @next_use.setter
    def next_use(self, value: str):
        """Сеттер на установку имени поля блокировки сессии

        :param value: имя поля
        """
        self._next_use = value

    @property
    def active(self) -> str:
        """Геттер на получение имени поля, отвечающего за полную блокировку сессии

        :return: имя поля
        """
        return self._active

    @active.setter
    def active(self, value: str):
        """Сеттер на установку имени поля полной блокировки сессии

        :param value: имя поля
        """
        self._active = value

    @property
    def session(self) -> str:
        """Геттер на получение имени поля, отвечающего за основные данные сессии

        :return: имя поля
        """
        return self._session

    @session.setter
    def session(self, value: str):
        """Сеттер на установку имени поля полной основных данных сессии

        :param value: имя поля
        """
        self._session = value

    @property
    def last_use(self) -> str:
        """Геттер на получение имени поля времени последнего использования сессии

        :return: имя поля
        """
        return self._last_use

    @last_use.setter
    def last_use(self, value: str):
        """Сеттер на установку имени поля времени последнего использования сессии

        :param value: имя поля
        """
        self._last_use = value

    @property
    def block_time(self) -> timedelta:
        """Геттер времени блокировки сессии

        :return: timedelta
        """
        return timedelta(**self._block_time)

    @block_time.setter
    def block_time(self, value: Period):
        """Сеттер на установку времени по умолчанию на блокировку сессии

        :param value: период блокировки, например, ``{'hours': 8}``, ``{'seconds': 120}``
        """
        self._block_time = value

    @property
    def lock_time(self) -> timedelta:
        """Геттер времени временной блокировки сессии

        :return: timedelta
        """
        return timedelta(**self._lock_time)

    @lock_time.setter
    def lock_time(self, value: Period):
        """Сеттер на установку времени по умолчанию на временную блокировку сессии

        :param value: период блокировки, например, ``{'hours': 8}``, ``{'seconds': 120}``
        """
        self._lock_time = value

    @property
    def default_filter(self) -> Dict:
        """Получение фильтра по умолчанию для операций выборки данных
        :return: dict фильтра
        """
        return self._default_filter

    @default_filter.setter
    def default_filter(self, filter_options: Dict):
        """Сеттер на установки фильтра по умолчанию

        :param value: период блокировки, например, ``{'hours': 8}``, ``{'seconds': 120}``
        """
        self._default_filter = filter_options

    @property
    def projection(self) -> Optional[Dict]:
        """Словарь для указания полей в выборке данных

        Аналог из SQL - ``select id, name from ... = {'id': 1, 'name': 1}``
        ``select * from ... = {}``

        :return: dict проекции
        """
        return self._projection

    @projection.setter
    def projection(self, options: Optional[Dict]):
        """Сеттер на установки проекции

        :param value: словарь, где ключи имена полей, значения 0 или 1
        """
        self._projection = options

    @property
    def next_use_delay(self) -> int:
        """Значение в секундах времени, на которое блокируется сессия сразу же после выборки ее из БД.

        Данное свойство необходимо для избежания ситуации, когда N обработчиков хотят
        параллельно брать одинаковую сессию. Сессия сразу же блокируется на M секунд.
        :return: время в сек
        """
        return self._next_use_delay

    @next_use_delay.setter
    def next_use_delay(self, value: int):
        """Сеттер установки моментальной блокировки сессии

        :param value: положительное число, в сек
        """
        self._next_use_delay = value if value >= 0 else 1
